techniques/applications/repoGpgKeyManagement/1.0/metadata.xml

```diff
@@ -0,0 +1,125 @@
+  <BUNDLES>
+    <NAME>check_repo_gpg_key</NAME>
+  </BUNDLES>
+
+  <!-- Define agent and OS compatibility for this Technique.
+     Currently, this is for information purposes only.
+     ## TODO: It has not been tested on any older OS, only on latest-greatest (SLES11, CentOS 7, Ubuntu 14)
+  <COMPATIBLE>
+    <OS version=">= 5">RHEL</OS>
+    <OS version=">= 11">SuSE Linux</OS>
+    <OS version=">= 4 (Etch)">Debian</OS>
+    <OS version=">= 5">RHEL / CentOS</OS>
+  </COMPATIBLE>
+-->
+
+  <TRACKINGVARIABLE>
+    <SAMESIZEAS>GPG_KEY_HASH</SAMESIZEAS>
+  </TRACKINGVARIABLE>
+
+  <!-- From here on, define variables to display in the web interface
+     All variables must be contained in a section.
+     Sections may be multivalued, or not -->
+ <SECTIONS>
+   <SECTION name="Repository GPG Key Management" multivalued="true" component="true" componentKey="GPG_KEY_HASH">
+     <SELECT1>
+       <NAME>GPG_KEY_ACTION</NAME>
+       <DESCRIPTION>Which operation should be done with this GPG Key</DESCRIPTION>
+       <ITEM>
+         <LABEL>Import (both hash and key content required)</LABEL>
+         <VALUE>add</VALUE>
+       </ITEM>
+       <ITEM>
+         <LABEL>Remove (only hash required)</LABEL>
+         <VALUE>del</VALUE>
+       </ITEM>
+       <CONSTRAINT>
+         <DEFAULT>add</DEFAULT>
+       </CONSTRAINT>
+     </SELECT1>
+     <INPUT>
+       <NAME>GPG_KEY_HASH</NAME>
+       <CONSTRAINT>
+         <MAYBEEMPTY>false</MAYBEEMPTY>
+         <REGEX error="Exactly 16 hexadecimal characters required"><![CDATA[ [a-fA-F0-9]{16} ]]></REGEX>
+       </CONSTRAINT>
+       <DESCRIPTION>Long hash of the GPG Key</DESCRIPTION>
+       <LONGDESCRIPTION>
+        You get it by looking for a line like `pub   2048R/70AF9E8139DB7C82 2013-01-31' when using thae command `gpg --list-keys --keyid-format=long'.
+        From that, fill in the 16 hexa-chars part like `70AF9E8139DB7C82' in the example.
+       </LONGDESCRIPTION>
+     </INPUT>
+     <INPUT>
+       <NAME>GPG_KEY_NAME</NAME>
+       <CONSTRAINT>
+         <MAYBEEMPTY>true</MAYBEEMPTY>
+         <REGEX error="No double quotation marks"><![CDATA[ [^"]* ]]></REGEX>
+       </CONSTRAINT>
+       <DESCRIPTION>Description</DESCRIPTION>
+       <LONGDESCRIPTION>This is only used to identify the Key in the GUI, it has no role in adding/removing the Key on the systems</LONGDESCRIPTION>
+     </INPUT>
+     <INPUT>
+       <NAME>GPG_KEY_CONTENT</NAME>
+       <CONSTRAINT>
+         <MAYBEEMPTY>true</MAYBEEMPTY>
+         <TYPE>textarea</TYPE>
+         <REGEX error="Need the whole part of PGP PUBLIC KEY BLOCK, as exported by `gpg -a --export'">
+           <![CDATA[ \s*-----BEGIN PGP PUBLIC KEY BLOCK-----[^'"\\]+-----END PGP PUBLIC KEY BLOCK-----\s* ]]>
+         </REGEX>
+       </CONSTRAINT>
+       <DESCRIPTION>The whole section of PGP PUBLIC KEY BLOCK</DESCRIPTION>
+       <LONGDESCRIPTION>Use the command `gpg -a --export' to export the public key</LONGDESCRIPTION>
```

Comment:
You could add something like "mandatory when adding a key and not used when removing one"